config/routes.rb

```diff
@@ -1,8 +1,15 @@
 Rails.application.routes.draw do
-  resources :users do
-    resources :adverts do
-      resources :comments
-    end
-  end
+  
+  devise_for :users,
+             controllers: { sessions: 'users/sessions', registrations: 'users/registrations'}
+  
+  patch '/users/set_admin/:id', to: 'users#create_admin' 
 
+  get '/adverts/unposted', to: 'adverts#unposted'
+  patch '/adverts/unposted/:id', to: 'adverts#approve'
```

Comment:
it would be better to define those two custom routes as
```ruby
resources :adverts do
   get :unposted, on: :collection
   post :unposted, on: :member
end
```
see: https://guides.rubyonrails.org/routing.html#adding-more-restful-actions